### PR TITLE
Fix timeline feature error when base url is set

### PR DESCRIPTION
**Pull request by Meriem-BenIsmail**
This PR fixes https://github.com/jupyterlab/jupyter-collaboration/issues/393

### CODE REVIEW

**Comment by davidbrochart** [comment]
Failures seem to be unrelated (see #403).
I will look at that.

**Review by davidbrochart — COMMENTED** [review]
Since #393 seems to suggest an issue with the server base URL, should you make use of it in this PR? The `baseUrl` is available in JupyterLab's page config.

**Comment by Meriem-BenIsmail** [comment]
> Since #393 seems to suggest an issue with the server base URL, should you make use of it in this PR? The `baseUrl` is available in JupyterLab's page config.

The problem isn't actually with the baseUrl, it's just that the document name was being incorrectly extracted.

**Comment by davidbrochart** [comment]
Could you write a test that shows that this was indeed the cause of problem? In particular, I'd like to see that your fix works even if "api/collaboration/timeline" is present in the base URL.

**Comment by Meriem-BenIsmail** [comment]
Currently, the updated logic ensures that only the last occurrence of `api/collaboration/timeline` is picked up, so it should handle the base URL scenario as expected. Given this, the fix seems okay for the mentioned issue.

Would you still like me to add a specific test to verify this behavior?

I also want to mention that there’s another open PR ([#401](https://github.com/jupyterlab/jupyter-collaboration/pull/401)) that creates a test file for the timeline slider. If I add a test here, it would create or modify the same file, potentially leading to conflicts later. Let me know how you'd like to proceed—whether to hold off on tests for this PR until #401 is merged or go ahead and add them now.

File: packages/docprovider/src/component.tsx
Comment (davidbrochart) [inline packages/docprovider/src/component.tsx:23]: We already have it [here](https://github.com/jupyterlab/jupyter-collaboration/blob/3126a81b78174cbc32c5f05d4678458f2d01f3eb/packages/docprovider/src/requests.ts#L15) and [here](https://github.com/jupyterlab/jupyter-collaboration/blob/3126a81b78174cbc32c5f05d4678458f2d01f3eb/packages/docprovider-extension/src/filebrowser.ts#L46). Should it be in one place?

**Comment by davidbrochart** [comment]
Yes it would be nice to have a test. I merged #401.

File: packages/docprovider/src/component.tsx
Comment (Meriem-BenIsmail) [reply packages/docprovider/src/component.tsx:23]: Yes I will fix that. Thanks for the feedback !

File: ui-tests/package.json
Comment (davidbrochart) [inline ui-tests/package.json:8]: You can get rid of `ui-tests/jupyter_server_test_config_timeline.py` and do:
```suggestion
    "start:timeline": "jupyter lab --ServerApp.base_url=/api/collaboration/timeline/",
```

File: ui-tests/tests/timeline-slider.spec.ts
Comment (davidbrochart) [inline ui-tests/tests/timeline-slider.spec.ts:34]: I don't understand what this test does. It seems that you do nothing if `isTimeline`?

File: ui-tests/tests/timeline-slider.spec.ts
Comment (davidbrochart) [inline ui-tests/tests/timeline-slider.spec.ts:50]: Same comment as above.

File: ui-tests/tests/timeline-slider.spec.ts
Comment (Meriem-BenIsmail) [reply ui-tests/tests/timeline-slider.spec.ts:34]: There is no need to execute this test in the added timeline test environment (with the set baseurl)
but this is the test for this PR that was merged https://github.com/jupyterlab/jupyter-collaboration/pull/401

File: ui-tests/tests/timeline-slider.spec.ts
Comment (davidbrochart) [inline ui-tests/tests/timeline-slider.spec.ts:44]:
```suggestion
```

File: ui-tests/tests/timeline-slider.spec.ts
Comment (davidbrochart) [inline ui-tests/tests/timeline-slider.spec.ts:32]: What is it for?

Comment (Meriem-BenIsmail) [reply ui-tests/tests/timeline-slider.spec.ts:32]: So I found that even with different configuration, somehow playwright overwrites the baseurl and goes to the default one (`http://localhost:8888`) by default so we need to use this `test.use({ autoGoto: false })`

Comment (Meriem-BenIsmail) [reply ui-tests/tests/timeline-slider.spec.ts:32]: isTimeline is the environment variable used to determine whether we are testing with the new environment or the default one. So we need to do this `test.use({ autoGoto: false })` only if we are testing with the new env (when `isTimeline` is true like in here `TIMELINE_FEATURE=1 jlpm test:timeline` )

Comment (davidbrochart) [reply ui-tests/tests/timeline-slider.spec.ts:32]: Could you explain what you mean by "new environment" and "default environment"?

Comment (Meriem-BenIsmail) [reply ui-tests/tests/timeline-slider.spec.ts:32]: The "new environment" refers to the testing setup, which uses a custom `baseUrl` (`/api/collaboration/timeline/`) as opposed to the "default environment," where tests run with the standard `baseUrl` (`/lab` or `/`). The new environment is specifically configured to validate the Timeline functionality in isolation, whereas the default environment is used for general tests without the Timeline-specific context.

`isTimeline` is used to identify which environment is active. If `isTimeline` is true, we disable the automatic navigation (`test.use({ autoGoto: false })`) that Playwright performs, as this navigation doesn’t respect the custom `baseUrl `and defaults to `http://localhost:8888`. Disabling it allows us to manually control the navigation to the appropriate custom environment.

**Review by davidbrochart — APPROVED** [review]
Thanls @Meriem-BenIsmail.

**Pull request merged**
